### PR TITLE
Fixed: Parser Improvements

### LINKED ISSUE #4198: Another obfuscated parser bug
**Describe the bug**
Another obfuscated case, causing the obfuscated title to be parsed as the season

**Screenshots**
See Reddit 

**Logs**
Not needed
Full File Path: /mnt/shared/config/sabnzbd/sabnzbd/complete/tv/SEAL.Team.S04E03.The.New.Normal.1080p.AMZN.WEB-DL.DDP5.1.H.264-NTb/cwnOJYks5E2WP7zGuzPkdkRK3JkWw0.mkv

**System Information**

Version
3.0.4.1026
Package Version
3.0.4.1026-ls13 by linuxserver.io
Mono Version
5.20.1.34

**UI Bugs:**
N/A

**Additional context**
Reported via [Reddit](https://www.reddit.com/r/sonarr/comments/km1puk/cute_little_bug/?utm_source=share&utm_medium=mweb)

### LINKED ISSUE #4199: Correctly detect "4Kto1080p" quality as 1080p
Currently Sonarr is incorrectly detecting "4Kto1080p" as 480p quality.

I only see this quality on "The Mandalorian" releases, but it'd be good if Sonarr could be fixed up to detect this properly as 1080p.

image

From the logs:
```
20-12-29 12:30:47.6|Debug|Parser|Parsing string 'The.Mandalorian.2019.S02E08.Chapter.16.The.Rescue.4Kto1080p.DSNYP.Webrip.x265.10bit.EAC3.5.1.Atmos.GokiTAoE'
20-12-29 12:30:47.6|Debug|Parser|Episode Parsed. The Mandalorian 2019 - S02E08 
20-12-29 12:30:47.6|Debug|Parser|Language parsed: English
20-12-29 12:30:47.7|Debug|QualityParser|Trying to parse quality for The.Mandalorian.2019.S02E08.Chapter.16.The.Rescue.4Kto1080p.DSNYP.Webrip.x265.10bit.EAC3.5.1.Atmos.GokiTAoE
20-12-29 12:30:47.7|Debug|Parser|Quality parsed: WEBRip-480p v1
```

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
Additional handling for various parser issues; see linked GHI

#### Todos
- [X] Tests
- Wiki Updates -N/A

#### Issues Fixed or Closed by this PR

* Fixes #4198
* Fixes #4199

### CODE REVIEW

File: src/NzbDrone.Core.Test/ParserTests/QualityParserFixture.cs [hunk 1]
```diff
@@ -240,6 +240,7 @@ public void should_parse_webdl1080p_quality(string title, bool proper)
         [TestCase("Better.Call.Saul.S03E09.1080p.NF.WEBRip.DD5.1.x264-ViSUM", false)]
         [TestCase("The Walking Dead S09E13 1.54 GB WEB-RIP 1080p Dual-Audio 2019 MKV", false)]
         [TestCase("Series.Title.1x04.ITA.1080p.WEBMux.x264-NovaRip", false)]
+        [TestCase("The.Mandalorian.2019.S02E07.Chapter.15.The.Believer.4Kto1080p.DSNYP.Webrip.x265.10bit.EAC3.5.1.Atmos.GokiTAoE", false)]
```
Comment (markus101) [inline src/NzbDrone.Core.Test/ParserTests/QualityParserFixture.cs:243]: Please replace the series title with the generic `Series.Title`

**Review by markus101 — CHANGES_REQUESTED** [review]
Please update the commit messages to close the individual issues they address.

The quality parsing change should be renamed to: 
Fixed: Parsing of 4Kto1080p as 1080p

**Review by markus101 — CHANGES_REQUESTED** [review]
he commits still need to be updated to include the GHIs they address.

**Comment by bakerboy448** [comment]
Just append the GHI # e.g  “(#XXXX)” to the commit messages, or what specifically are you looking for?

**Comment by markus101** [comment]
Yeah the typical format is:

```
Fixed: xyz

Closes #n
```

Example commit: https://github.com/Sonarr/Sonarr/commit/e5e86680c8f99687d3b95765be1c414cd1090191

**Comment by bakerboy448** [comment]
@markus101 should be good now!  Learn something new every PR :D

**Pull request merged**